docs/cluster_api.md

```diff
@@ -0,0 +1,95 @@
+# Cluster Operator API
+
+## Overview
+
+The cluster operator API defines a set of resources that allow the instantiation of OpenShift clusters and subsequent
+updates to those clusters based on changes to said resources.
+
+The general approach for these resources is to follow the Kubernetes model of a spec section that is end-user controlled and
+a status section that represents the current state of the resources. Changes to the spec will result in the necessary actions
+by the system to reconcile the current state of the system with what the user has specified.
+
+## Resources
+
+### Cluster
+
+The cluster resource should allow 2 levels/scopes of spec:
+- Cluster-wide configuration
+- NodeGroup configuration
+
+Furthermore, each scope can be described in 2 parts:
+- Hardware spec (needed to instantiate the proper virtual hardware)
+- OpenShift config (openshift-specific configuration)
+
+For the Status part, if the cluster has been instantiated, we need to know how to access it (Kubeconfig may be sufficient)
+
+In more detail:
+
+#### Spec:
+* Cluster-wide configuration:
+  * ObjectMetadata (inline ObjectMetadata)
+  * TypeMetadata (inline TypeMetadata)
+  * Hardware-spec
+    * Type (AWS, Azure, GCE, etc) [string]
+    * AWSConfig: [struct]
+      - AWSSecret [LocalReference]
+      - Region [string]
+      - VPCName [string]
+      - VPCSubnet [string]
```

Comment:
I'm told that one of these days we'll automate building a multi-AZ clusters, so we likely want VPCSubnets (plural) so we can have an AZ-aware list of subnets that will be attached to the VPC

Reply:
thx @joelddiaz. I guess that means we can't derive the AZ subnets from the main VPC ? (this made me think we could http://pastebin.test.redhat.com/541799 - see highlighted). If not, I guess we want a map of AZ->subnet, correct?

Reply:
We pretty much carve up the subnets however we want. In practice we have a defined list of regions/subnet-lists that we use in case the specific cluster doesn't have any particular requirements, but in the case where we need to VPC pair with a customer's existing VPC, we massage these values as needed.